src/app/Library/Auth/AuthenticatesUsers.php

```diff
@@ -47,6 +48,18 @@ public function login(Request $request)
         }
 
         if ($this->attemptLogin($request)) {
+            if (config('backpack.base.setup_email_verification_routes', false) ) {
+                $user = $this->guard()->user();
+                if ($user->email_verified_at) {
+                    return $this->sendLoginResponse($request);
+                } else {
+                    $this->guard()->logout();
+                    Cookie::queue('backpack_email_verification', $user->{config('backpack.base.email_column')}, 30);
+                    return $request->wantsJson()
+                        ? new Response('Email verification required', 403)
+                        : redirect(route('verification.notice'));
+                }
+            }
```

Comment:
Let's move all of this to a separate method. Cleaner that way - we can return early a bunch of times so it'll be easier to read.